src/pybind/mgr/smb/tests/test_fs.py

```diff
@@ -65,3 +65,34 @@ def mmcmd(cmd):
     )
     with pytest.raises(FileNotFoundError):
         fspr.resolve_exists('cephfs', 'alpha', 'beta', '/zowie')
+
+
+def test_caching_fs_path_resolver(monkeypatch):
+    monkeypatch.setattr('cephfs.ObjectNotFound', KeyError)
+
+    def mmcmd(cmd):
+        if cmd['prefix'] == 'fs subvolume getpath':
+            if cmd['vol_name'] == 'cached_cephfs' and cmd['sub_name'] == 'cached_beta':
+                return 0, '/volumes/cool/cached_path/f00d-600d', ''
+        return -5, '', 'cached_eek'
+
+    m = mock.MagicMock()
+    m.mon_command.side_effect = mmcmd
+
+    fspr = smb.fs.AdvancedCachingCephFSPathResolver(m)
+
+    # Resolve a path (cache miss)
+    path = fspr.resolve('cached_cephfs', 'cached_alpha', 'cached_beta', '/cached_zowie')
+    assert path == '/volumes/cool/cached_path/f00d-600d/cached_zowie'
+
+    # Resolve the same path again (cache hit)
+    path = fspr.resolve('cached_cephfs', 'cached_alpha', 'cached_beta', '/cached_zowie')
+    assert path == '/volumes/cool/cached_path/f00d-600d/cached_zowie'
```

Comment:
To validate the hits & misses we can use the mock object's call counters.

Reply:
Done!